docs/brokers.rst

```diff
@@ -96,5 +105,131 @@ example:
+library, a Kafka client written in Python.
+
+Adding a Kafka Event Broker Using the Endpoint Configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+As for the other brokers, you can use an endpoint configuration file to instruct Rasa Core to stream
+all events to this event broker. To do it, add the following section to your
+endpoint configuration.
+
+Pass the ``endpoints.yml`` file as argument with ``--endpoints <path to your endpoint configuration>``
+when running Rasa, and select Kafka producer with ``--kafka_broker kafka_producer``, as following example:
+
+.. code-block:: shell
+
+    python3 -m rasa_core.run -d models/dialogue -u models/nlu/current --event_broker kafka_producer --endpoints endpoints.yml
+
+Using SASL_PLAINTEXT protocol the endpoints file must have the following entries:
+
+.. literalinclude:: ../data/test_endpoints/event_brokers/kafka_plaintext_endpoint.yml
+
+In the case of using SSL protocol the endpoints file must looks like:
+
+.. literalinclude:: ../data/test_endpoints/event_brokers/kafka_ssl_endpoint.yml
+
+Adding a Kafka Broker in Python
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The code below shows an example on how to instantiate a Kafka producer in you script.
+
+.. code-block:: python
+
+    from rasa_core.event_brokers.kafka_producer import KafkaProducer
+    from rasa_platform.core.tracker_store import InMemoryTrackerStore
+
+    kafka_broker = KafkaProducer(host='localhost:9092',
+                                 topic='rasa_core_events')
+
+    tracker_store = InMemoryTrackerStore(event_broker=kafka_broker)
+
+
+The host variable can be either a list of brokers adresses or a single one.
+If only one broker address is available, the client will connect to it and 
+request the cluster Metadata.
+Therefore, the remain brokers in the cluster can be discovered
+automatically through the data served by the first connected broker.
+
+To pass more than one broker address as argument, they must be passed in a
+list of strings. e.g.:
+
+.. code-block:: python
+
+    kafka_broker = KafkaProducer(host=['kafka_broker_1:9092',
+                                       'kafka_broker_2:2030',
+                                       'kafka_broker_3:9092'],
+                                 topic='rasa_core_events')
+
+Authentication and authorization
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Rasa Core's Kafka producer accepts two types of security protocols - SASL_PLAINTEXT and SSL.
+
+For development environment, or if the brokers servers and clients are located
+into the same machine, you can use simple authentication with SASL_PLAINTEXT.
+By using this protocol, the credentials and messages exchanged between the clients and servers
+will be sent in plaintext. Thus, this is not the most secure approach, but since it's simple
+to configure, it is useful for simple cluster configurations.
+SASL_PLAINTEXT protocol requires the setup of the 'username' and 'password'
+previously configured in the broker server.
```

Comment:
what do you mean here, where is this configured? I guess in the running Kafka instanc, right?

Reply:
It's configured on both client and server. There is communication between client<->server and between servers itself on cluster. The security protocol being used may differ for these two communications.

This is referred to producer(client) <-> server communication.